### DISCUSSION TITLE
[flutter] multiple user login and store session

### DISCUSSION THREAD

**Discussion by Gerald-Gmainer** [supabase/supabase#13983] [Questions]
Hi, my client wants to use a tablet, where multiple uses can to login quickly. The users should be stored in the device. By clicking an user from a list, the app should login (without entering email/password).

My idea was to store the session
`supabase.auth.currentSession.persistSessionString`
to local storage.

And use
`await supabase.auth.recoverSession(storedSessionString)`
to login that user.

Which works, until the token is expired and get this error

```
E/flutter ( 7039):  AuthException(message: Invalid Refresh Token: Refresh Token Not Found, statusCode: 400)
E/flutter ( 7039): #0      GotrueFetch.request (package:gotrue/src/fetch.dart:99:7)
E/flutter ( 7039): <asynchronous suspension>
E/flutter ( 7039): #1      GoTrueClient._callRefreshToken (package:gotrue/src/gotrue_client.dart:669:24)
E/flutter ( 7039): <asynchronous suspension>
E/flutter ( 7039): #2      GoTrueClient.recoverSession (package:gotrue/src/gotrue_client.dart:549:26)
E/flutter ( 7039): <asynchronous suspension>
E/flutter ( 7039): #3      _SettingTabState._loadUser (package:flutter_app/presentation/tab/setting/setting_tab.dart:199:29)
E/flutter ( 7039): <asynchronous suspension>

```
Am I am doing something wrong? Is that workflow even possible?

**Comment by GaryAustin1** [comment]
You can also have separate clients per user (set unique storage key name in createClient. Then the session data is stored separately for you.

**Reply by Gerald-Gmainer** [reply]
Thanks, but not sure how do that. The supabase flutter client has no storage key, but uses a localStorage parameter (hivebox)

I wrote my custom localStorage hivebox, but I dont know to re-initialize
`await Supabase.initialize(url: url,anonKey: anonKey, localStorage: customStorage);`

`Unhandled Exception: 'package:supabase_flutter/src/supabase.dart': Failed assertion: line 76 pos 7: '!_instance._initialized': This instance is already initialized`

**Reply by GaryAustin1** [reply]
Sorry, I assumed the init had similar options to js with a key.
I don't see anything in the source or docs about it.  
Closest comment:
https://github.com/supabase/supabase-flutter/issues/345

With Supabase-js it does work.    Y

**Comment by dshukertjr** [comment]
I like your approach of storing customer sessions in separate places and alternating between them. supabase_flutter allows you to define your own custom storage to store and retrieve session data, so you could probably create your own that takes care of switching between users.
https://github.com/supabase/supabase-flutter/blob/main/lib/src/local_storage.dart
